recipes/mp-units/0.7.0/conanfile.py

```diff
@@ -1,70 +1,121 @@
-from conans import ConanFile, CMake, tools
-from conans.tools import Version, check_min_cppstd
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=2.0.0"
+
 
 class MPUnitsConan(ConanFile):
     name = "mp-units"
     homepage = "https://github.com/mpusz/units"
-    description = "Physical Units library for C++"
-    topics = ("units", "dimensions", "quantities", "dimensional-analysis", "physical-quantities", "physical-units", "system-of-units", "cpp23", "cpp20", "library", "quantity-manipulation")
+    description = "Physical Quantities and Units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake_find_package_multi"
+    requires = ["gsl-lite/0.38.0", "fmt/7.1.3"]
     no_copy_source = True
+    generators = "CMakeToolchain", "CMakeDeps"
 
```

Comment:
I think you meant `package_type = "header-library"`? Also, do I need to provide `package_id()` in such case?

Reply:
Sorry, you're correct. It's `header-library`. I'm not actually sure if you'd still need to include the `package_id()` method in this case.

Reply:
OK, I removed `package_id()` for now but I can easily restore it if needed.

Reply:
I checked in conan-io/conan#13289 on this and the `package_id()` method is still required since `package_type` only effects consumers of the package and not the package directly.

Reply:
OK, `package_id()` is restored now.